windows/src/desktop/kmshell/util/Keyman.System.KeymanStartTask.pas

```diff
@@ -144,31 +153,44 @@ class procedure TKeymanStartTask.DeleteTask;
         if E.ErrorCode = HResultFromWin32(ERROR_FILE_NOT_FOUND) then
           // Assume that the folder doesn't exist, nothing to do
           Exit;
+        raise;
       end;
     end;
 
     try
-      pTaskFolder.DeleteTask(CTaskName, 0);
+      pTaskFolder.DeleteTask(GetTaskName, 0);
     except
       on E:EOleException do
       begin
-        if E.ErrorCode = HResultFromWin32(ERROR_FILE_NOT_FOUND) then
-          // Assume that the task doesn't exist, nothing to do
-          Exit;
+        if E.ErrorCode <> HResultFromWin32(ERROR_FILE_NOT_FOUND) then
+          // We ignore when the task doesn't exist, but report other errors
+          TKeymanSentryClient.ReportHandledException(E, 'Failed to delete task '+GetTaskName);
```

Comment:
Why not follow your pattern ll. 153-156? If you don't raise here, will it ever get to l. 216?

Reply:
Here, we want to continue so that we attempt to delete the folder that contains the task, even if the task does not exist. Without re-raising, it will not drop through to the outer handler at l. 216, no. That's intentional.